check/src/unify_type.rs

```diff
@@ -180,16 +182,22 @@ fn do_zip_match<'a, U>(self_: &ArcType,
                 }
                 Less => {
                     let offset = r_args.len() - l_args.len();
-                    let new_type =
-                        unifier.try_match(l, &Type::app(r.clone(), r_args[..offset].into()));
+
+                    let reduced_r = Type::app(r.clone(),
+                                              r_args[..offset].iter().cloned().collect());
```

Comment:
Agreed...  no real 'best way' here. Not a big deal - I agree it might be a tad superfluous here.

_shakes fist at Rust's ugly syntax and rightward drift_

😆